### DISCUSSION TITLE
Figures from notebook outputs with link to img file

### DISCUSSION THREAD

**Discussion by whitead** [executablebooks/meta#114] [Q&A] [answered]
I have a notebook with a complex image output. It would be nice to make the image be link so that a mobile user could click to see the image. I know that [normal figures are links](https://jupyterbook.org/reference/cheatsheet.html#figures-and-images).  Normal output from notebooks do not have links. I tried using `glue` and then [a figure directive](https://jupyterbook.org/content/glue.html#the-glue-figure-directive), but that does not provide a link. Is there a way I can change the directive or must I save the figure and link to it? Thank you!

**Comment by whitead** [comment]
I wrote some javascript that can be included. Maybe this will help someone else. Seems important on mobile especially. Put the code below in `_static/pick_a_name.js`.

```js

function insert_anchor(element) {
    if (element.parentElement.tagName !== 'A') {
        let new_a = document.createElement('a')
        new_a.href = element.getAttribute('src')
        let p = element.parentElement
        element.parentElement.removeChild(element)
        new_a.appendChild(element)
        p.appendChild(new_a)
    }
}

function add_img_anchors() {
    let figs = document.querySelectorAll(".figure img")
    figs.forEach(insert_anchor)
    let cell_outputs = document.querySelectorAll(".cell_output img")
    cell_outputs.forEach(insert_anchor)
}

window.addEventListener('load', **add_img_anchors)**
```

**Reply by chrisjsewell** [reply]
Thanks for sharing @whitead 😄
Yeh I see what you mean, I will try to look into this in myst-nb at some point, it should certainly be possible

**Reply by chrisjsewell** [reply]
I've opened https://github.com/executablebooks/MyST-NB/issues/240

**Comment by chrisjsewell** [comment, marked as answer]
New in v0.8.0 see https://jupyterbook.org/content/figures.html#markdown-figures 😄

**Reply by whitead** [reply]
Thanks @chrisjsewell !

**Reply by chrisjsewell** [reply]
BTW I noted a weird thing happening, whereby the image/figure would only be a link, if it included a height/width. Don't know if this was just for me, or is a bug in sphinx, myst etc

**Reply by whitead** [reply]
@chrisjsewell I think it's always been the case. At least since I've been using jupyter book.
